src/ClientData/CallstackDataTest.cpp

```diff
@@ -200,4 +200,124 @@ TEST(CallstackData, FilterCallstackEventsBasedOnMajorityStartExcludesFunctionToS
                                                                          event4, event5, event6}));
 }
 
+TEST(CallstackData, ForEachCallstackEventOfTidInTimeRangeDiscretized) {
+  CallstackData callstack_data;
+
+  const uint64_t cs_id = 12;
+  CallstackInfo cs{{0x11, 0x10}, CallstackType::kComplete};
+  callstack_data.AddUniqueCallstack(cs_id, std::move(cs));
+
+  constexpr uint32_t kTid = 42;
+  constexpr uint32_t kAnotherTid = 43;
+
+  const uint64_t time1 = 142;
+  CallstackEvent event1{time1, cs_id, kTid};
+  callstack_data.AddCallstackEvent(event1);
+
+  const uint64_t time2 = 242;
+  CallstackEvent event2{time2, cs_id, kTid};
+  callstack_data.AddCallstackEvent(event2);
+
+  const uint64_t time3 = 342;
+  CallstackEvent event3{time3, cs_id, kAnotherTid};
+  callstack_data.AddCallstackEvent(event3);
+
+  const uint64_t time4 = 442;
+  CallstackEvent event4{time4, cs_id, kTid};
+  callstack_data.AddCallstackEvent(event4);
```

Comment:
A suggestion. You need to construct 4 CallstackEvents. So another option is to hard-code two lists of 4 elements -- `kTids` and `kTimes`. Then you can `absl::c_transfrom` the two lists and yield a collection of CallstackEvent.  And finally, populate `callstack_data` by a simple loop over the collection. This would cut code duplication. And all of these can be global variables.

Reply:
This is what I had in mind:

```
const std::vector<uint32_t> kTids = {kTid, kTid, kAnotherTid, kTid};
const std::vector<uint64_t> kTimestamps = {142, 242, 342, 442};

const std::vector<CallstackEvent> kAllEvents = [] {
  std::vector<CallstackEvent> events;
  absl::c_transform(kTids, kTimestamps, std::back_inserter(events),
                    [](uint32_t tid, uint64_t timestamp) {
                      return CallstackEvent{timestamp, kCallstackId1, tid};
                    });
  return events;
}();
```

This way way we don't have to introduce a named function, call it multiple times, introduce the `TidAndTimestamp` struct. 

OTOH, feel free to keep your solution. Just rename the vector `kTimestampsAndTidInForEachCallstackTests`. I think, `kTidAndTimestampInForEachCallstackTest` suits it better.

Reply:
The TidAndTimestamp struct was created just to assure that the 2 vectors have the same size. But TBH I don't have a strong opinion about that, I can divided between 2 different vectors since it's a test.

Reply:
You can make sure that the length is the same by using `std::array` instead of `std::vector` for `kTids` and `kTimestamps`.

Reply:
Yes, sure. I will keep vectors as it is a test.